### PR TITLE
Support configurable extra fields for LazyNeMoTarredIterator

### PR DESCRIPTION
**Pull request by pzelasko**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/common/data/lhotse/nemo_adapters.py [hunk 1]
```diff
@@ -274,10 +322,13 @@ def shard_ids(self) -> List[int]:
     def __iter__(self) -> Generator[Cut, None, None]:
         shard_ids = self.shard_ids
 
+        seed = resolve_seed(self.shard_seed)
         if self.shuffle_shards:
-            seed = resolve_seed(self.shard_seed)
             random.Random(seed).shuffle(shard_ids)
 
+        # Propagate the random seed
```
Comment (krishnacpuvvada) [inline nemo/collections/common/data/lhotse/nemo_adapters.py:329]: just for reproducibility or is there any other reason?

File: nemo/collections/common/data/lhotse/nemo_adapters.py [hunk 2]
```diff
@@ -56,16 +59,33 @@ class LazyNeMoIterator:
...
 
         >>> cuts = lhotse.CutSet(LazyNeMoIterator("nemo_manifests/train.json"))
+
+    We allow attaching custom metadata to cuts from files other than the manifest via ``extra_fields`` argument.
+    In the example below, we'll iterate file "questions.txt" together with the manifest and attach each line
+    under ``cut.question`` using the field type ``text_iter``::
+
+        >>> cuts = lhotse.CutSet(LazyNeMoIterator(
+        ...     "nemo_manifests/train.json",
+        ...     extra_fields=[{"type": "text_iter", "name": "question", "path": "questions.txt"}],
```
Comment (krishnacpuvvada) [inline nemo/collections/common/data/lhotse/nemo_adapters.py:69]: is there value in allowing to path to be a list of questions also ?

File: nemo/collections/common/data/lhotse/nemo_adapters.py (hunk 1 above)
Comment (pzelasko) [reply nemo/collections/common/data/lhotse/nemo_adapters.py:329]: Reproducibility/consistent RNG behavior across all dataloading modules.

File: nemo/collections/common/data/lhotse/nemo_adapters.py (hunk 2 above)
Comment (pzelasko) [reply nemo/collections/common/data/lhotse/nemo_adapters.py:69]: I'm not sure I understood your comment - do you mean `"path": ["questions1.txt", "questions2.txt"]`? what would be the expected behavior?

**Review by zhehuaichen — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**